src/Native/Unix/System.Net.Security.Native/pal_gssapi.c

```diff
@@ -242,49 +223,31 @@ uint32_t NetSecurityNative_InitSecContext(uint32_t* minorStatus,
         gssCbt.application_data.value = cbt;
     }
 
-    GssName* targetName;
-    if (isNtlm || isNtlmFallback)
+    uint32_t majorStatus = gss_init_sec_context(minorStatus,
+                                                claimantCredHandle,
+                                                contextHandle,
+                                                targetName,
+                                                desiredMech,
+                                                reqFlags,
+                                                0,
+                                                (cbt != NULL) ? &gssCbt : GSS_C_NO_CHANNEL_BINDINGS,
+                                                &inputToken,
+                                                &outmech,
+                                                &gssBuffer,
+                                                retFlags,
+                                                NULL);
+
+    if (isNtlm)
```

Comment:
Seems like this could be:
```C#
*isNtlmUsed = (isNtlm || majorStatus != GSS_S_COMPLETE || gss_oid_equal(outmech, krbMech) == 0) ? 1 : 0;
```
?

Reply:
This is C code and not C#.  Does the tertiary operation work in C? :)

Reply:
Yes, C has a ternary operator.

Reply:
Thank you!  I didn't realize that.  I'll make the changes you suggested.